### PR TITLE
Adding some liquid support

### PR DESCRIPTION
**Pull request by wAsnk**
As the title says, exposed some viewmodels to liquid templates and added a filter that converts the JToken into the Amount struct. It's needed for liquid template overrides.

### CODE REVIEW

**Comment by wAsnk** [comment]
@DAud-IcI or @DemeSzabolcs could you review this?

File: src/Modules/OrchardCore.Commerce/Startup.cs [hunk 1]
```diff
@@ -143,6 +147,18 @@ public override void ConfigureServices(IServiceCollection services)
...
+        services.Configure<TemplateOptions>(option =>
+        {
+            option.MemberAccessStrategy.Register<ShoppingCartViewModel>();
+            option.MemberAccessStrategy.Register<ShoppingCartCellViewModel>();
+            option.MemberAccessStrategy.Register<ShoppingCartLineViewModel>();
+            option.MemberAccessStrategy.Register<Amount, string>((obj, _) => obj.ToString());
+            option.MemberAccessStrategy.Register<Amount, decimal>((obj, _) => obj.Value);
+        })
+        // Liquid filter to convert JToken value to Amount struct in liquid.
+        .AddLiquidFilter<ToAmountConverterFilter>("toAmount");
```
Comment (DAud-IcI) [inline src/Modules/OrchardCore.Commerce/Startup.cs:161]: Liquid filters are supposed to be snake_case ([for example](https://docs.orchardcore.net/en/dev/docs/reference/modules/Liquid/#absolute_url)). I think for conversion filters you are also just supposed to name it by the type, so `amount`.
Also please indent this.

File: src/Modules/OrchardCore.Commerce/Liquid/ToAmountConverterFilter.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace OrchardCore.Commerce.Liquid;
+
+public class ToAmountConverterFilter : ILiquidFilter
+{
+    public ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, LiquidTemplateContext context)
+    {
+        if (input.Type == FluidValues.Nil) return new ValueTask<FluidValue>(input);
+
+        var newObject = (input.ToObjectValue() as JToken).ToObject<Amount>();
```
Comment (DAud-IcI) [inline src/Modules/OrchardCore.Commerce/Liquid/ToAmountConverterFilter.cs:16]: `as` can return `null` if the type doesn't match, so this filter will throw NRE if used on the wrong input.

**Review by DAud-IcI — CHANGES_REQUESTED** [review]

File: src/Modules/OrchardCore.Commerce/Liquid/AmountConverterFilter.cs [hunk 3]
```diff
@@ -7,13 +7,17 @@
...
 namespace OrchardCore.Commerce.Liquid;
 
-public class ToAmountConverterFilter : ILiquidFilter
+public class AmountConverterFilter : ILiquidFilter
 {
     public ValueTask<FluidValue> ProcessAsync(FluidValue input, FilterArguments arguments, LiquidTemplateContext context)
     {
-        if (input.Type == FluidValues.Nil) return new ValueTask<FluidValue>(input);
+        if (input.Type == FluidValues.Nil ||
+            input.ToObjectValue() as JToken is not { } newInput)
```
Comment (DAud-IcI) [inline src/Modules/OrchardCore.Commerce/Liquid/AmountConverterFilter.cs:15]:
```suggestion
            input.ToObjectValue() is not JToken newInput)
```

**Pull request merged**